### DISCUSSION TITLE
<FATAL>: fail to mount after 10 seconds, please mount in foreground

### DISCUSSION THREAD

**Discussion by ahnselina** [juicedata/juicefs#1092] [Q&A] [answered]
I compiled juicefs-0.17.2 (added some codes)can run in foreground, but run with adding the "-d" always get error “<FATAL>: fail to mount after 10 seconds, please mount in foreground” , then I trace it use dlv, found that normal case without "-d" the info
```
io/fs.FileInfo(*os.fileStat) *{
	name: "test",
	size: 4096,
	mode: 2147484159,
	modTime: time.Time {
		wall: 237256785,
		ext: 63774031933,
		loc: *(*time.Location)(0x47161a0),},
	sys: syscall.Stat_t {
		Dev: 48,
		Ino: 1,
		Nlink: 3,
		Mode: 16895,
		Uid: 0,
		Gid: 0,
		X__pad0: 0,
		Rdev: 0,
		Size: 4096,
		Blksize: 65536,
		Blocks: 8,
		Atim: (*syscall.Timespec)(0xc0005a8150),
		Mtim: (*syscall.Timespec)(0xc0005a8160),
		Ctim: (*syscall.Timespec)(0xc0005a8170),
		X__unused: [3]int64 [0,0,0],},}
```
and the abnormal case with "-d"，I got the info:
```
io/fs.FileInfo(*os.fileStat) *{
	name: "test",
	size: 4096,
	mode: 2147484141,
	modTime: time.Time {
		wall: 964481087,
		ext: 63774010731,
		loc: *(*time.Location)(0x47161a0),},
	sys: syscall.Stat_t {
		Dev: 2049,
		Ino: 150765,
		Nlink: 2,
		Mode: 16877,
		Uid: 0,
		Gid: 0,
		X__pad0: 0,
		Rdev: 0,
		Size: 4096,
		Blksize: 4096,
		Blocks: 8,
		Atim: (*syscall.Timespec)(0xc000a88cb0),
		Mtim: (*syscall.Timespec)(0xc000a88cc0),
		Ctim: (*syscall.Timespec)(0xc000a88cd0),
		X__unused: [3]int64 [0,0,0],},}
```
in the cmd/mount_unix.go:43
```
                         if sys, ok := st.Sys().(*syscall.Stat_t); ok && sys.Ino == 1 {
				logger.Infof("\033[92mOK\033[0m, %s is ready at %s", name, mp)
				return
			}
```

``` sys.Ino == 1 ``` why must  ```sys.Ino == 1 ```?
in the abormal case, the sys.Ino != 1, so can't enter the right exec path. I want to know what's is meaning of ```sys.Ino == 1```

**Comment by xiaogaozi** [comment, marked as answer]
The inode 1 is the root directory of JuiceFS, you could check the information with `juicefs info` command.

**Reply by ahnselina** [reply]
> The inode 1 is the root directory of JuiceFS, you could check the information with `juicefs info` command.

In other words, only if this condition inode == 1 is met, the juicefs mount is ready?
In my understanding, the inode == 1 means the dir is the mount point.

**Reply by xiaogaozi** [reply]
When the inode of mount point is equal to 1, it means JuiceFS mount successfully.

**Reply by ahnselina** [reply]
OK，I see, thanks.
